scripts/dev/e2e.py

```diff
@@ -222,7 +218,13 @@ def main():
 
     pod = create_test_runner_pod(args.test)
     corev1 = client.CoreV1Api()
-    wait_for_pod_to_be_running(corev1, TEST_RUNNER_NAME, dev_config.namespace)
+
+    print("Waiting for pod to be running")
+    if not wait_for_condition(
+            lambda: corev1.read_namespaced_pod(TEST_RUNNER_NAME, dev_config.namespace),
+            lambda pod : pod.status.phase=="Running",
+            sleep_time=5, timeout=50, exceptions_to_ignore=ApiException):
+        raise Exception("Pod never got into Running state!")
```

Comment:
what do you think of keeping the function `wait_for_pod_to_be_running` but just change the implementation to use `wait_for_condition`. I think we get better readability this way instead of needing to parse the different lambdas.

Reply:
Makes sense and makes it clearer, updated!